File: PRAK605/PRAK605_2310817110011_Ghani-Mudzakir.py
ordo = int(input())

matriksA = [[0] * ordo for _ in range(ordo)]
matriksB = [[0] * ordo for _ in range(ordo)]
matriksHasil = [[0] * ordo for _ in range(ordo)]

# Input matriks A
print("Matriks A")
for i in range(ordo):
    input_row = input().split()
    matriksA[i] = [int(elem) for elem in input_row]

print("Matriks B")
for i in range(ordo):
    input_row = input().split()
    matriksB[i] = [int(elem) for elem in input_row]

# kita ulah matriks nya jadi 0 dhulu
for i in range(ordo):
    for j in range(ordo):
        matriksHasil[i][j] = 0

# hitung hitungan matriks nya
for i in range(ordo):
    for j in range(ordo):
        for k in range(ordo):
            matriksHasil[i][j] += matriksA[i][k] * matriksB[k][j]

# Print hasil
print("Matriks AXB\n")
for i in range(ordo):
    for j in range(ordo):
        print(matriksHasil[i][j], end=" ")
    print()

File: PRAK605/PRAK605_2310817110011_Ghani-Mudzakir.c
#include <stdio.h>

int main()
{
    int ordo;
    scanf("%d", &ordo);

    int matriksA[ordo][ordo], matriksB[ordo][ordo], matriksHasil[ordo][ordo];

    printf("Matriks A\n");
    for (int  i = 0; i < ordo; i++)
    {
        for (int j = 0; j < ordo; j++)
        {
            scanf("%d", &matriksA[i][j]);
        }

    }

    printf("Matriks B\n");
    for (int  i = 0; i < ordo; i++)
    { 
        for (int j = 0; j < ordo; j++)
        {
            scanf("%d", &matriksB[i][j]);
        }
    }

    // kita ulah matriks hasil nya jadi 0 dhulu
    for (int i = 0; i < ordo; i++)
    {
        for (int j = 0; j < ordo; j++)
        {
            matriksHasil[i][j] = 0;
        }
    }

    // kita ulah perkalian matriksnya
    for (int i = 0; i < ordo; i++)
    {
      for (int j = 0; j < ordo; j++)
        {
            for (int k = 0; k < ordo; k++)
            {
                matriksHasil[i][j] += (matriksA[i][k]*matriksB[k][j]);
            }   
        }
    }

    // coba kita print
    printf("Matriks AXB\n");
    for (int i = 0; i < ordo; i++)
    {
      for (int j = 0; j < ordo; j++)
        {
           printf("%d ", matriksHasil[i][j]);
        }
        printf("\n");
        
    }
    
    
    return 0;
}



